WORKSPACE

```diff
@@ -80,3 +80,6 @@ filegroup(
 )
 """
 )
+
+# we load toolchain that fails the build on unused warnings
+register_toolchains("//test_expect_failure/scalacopts_from_toolchain:failing_scala_toolchain")
```

Comment:
does this need to be in the WORKSPACE? Is that the only place to register new toolchains?

Is it the case that you can only have 1 version of each toolchain in play in a given repo? Is that how it works?

Reply:
according to https://docs.bazel.build/versions/master/toolchains.html#overview
> When a target requests a toolchain, Bazel checks the list of registered toolchains and creates a dependency from the target to the first matching toolchain it finds. To find a matching toolchain, Bazel does the following:
> 
> Looks through the registered toolchains, first from the --experimental_extra_toolchains flag, then from the registered_toolchains calls in the project's WORKSPACE file.
> 
> For each registered toolchain, Bazel performs the following checks:
> 
> a. Does the toolchain match the requested toolchain type? If not, skip it.
> 
> b. Do the toolchain's execution and target constraints match the constraints stated in the project's execution and target platforms? If yes, the toolchain is a match.
>
> **Because Bazel always selects the first matching toolchain, order the toolchains by preference if you expect the possibility of multiple matches.**

Reply:
You can register toolchains via a flag, or in the WORKSPACE. First match is selected.

Reply:
I just think it is going to confuse me and others that the only registered toolchain is on a path with "expect_failure" which just does not seem correct.

Reply:
I agree and I think this is because we couldn't utilize bazel e2e testing :(
You do understand that this is the toolchain for a test right? It's not what users are supposed to use.
@or-shachar is there one that users are supposed to use?
@johnynek maybe making the comment clearer it's for testing?
@or-shachar why not register it via a command line flag of the specific test?